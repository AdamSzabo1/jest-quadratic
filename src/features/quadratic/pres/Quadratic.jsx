
import * as style from './quadratic.module.css';

const Quadratic = () => {

    return (
        <div className={`${style.quadratic}`}>
            <h1 data-testid="quadratic-heading">Quadratic Equation</h1>
        </div>
    );
}

export default Quadratic;
