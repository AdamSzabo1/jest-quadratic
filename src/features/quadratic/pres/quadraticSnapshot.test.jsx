import React from "react";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';

import Quadratic from "./Quadratic";

describe('Quadratic snapshot', () => {

    test('should match previous snapshot with KaTeX', () => {

        const { asFragment } = render(
            <Quadratic
                a={1}
                b={8}
                c={-48}
                D={256}
                x1={4}
                x2={-12}
            />
        );

        expect( asFragment() ).toMatchSnapshot();
    });

});
