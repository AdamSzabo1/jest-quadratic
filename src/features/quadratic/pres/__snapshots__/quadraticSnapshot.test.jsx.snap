// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quadratic snapshot should match previous snapshot with KaTeX 1`] = `
<DocumentFragment>
  <div
    class="undefined"
  >
    <h1
      data-testid="quadratic-heading"
    >
      Quadratic Equation
    </h1>
    <div
      class="undefined"
    >
      <form>
        <h2>
          Input
        </h2>
        <div>
          <label
            for="coe-a"
          >
            a: 
          </label>
          <input
            data-testid="coe-a"
            id="coe-a"
            name="a"
            type="number"
            value="1"
          />
        </div>
        <div>
          <label
            for="coe-b"
          >
            b: 
          </label>
          <input
            data-testid="coe-b"
            id="coe-b"
            name="b"
            type="number"
            value="8"
          />
        </div>
        <div>
          <label
            for="coe-c"
          >
            c: 
          </label>
          <input
            data-testid="coe-c"
            id="coe-c"
            name="c"
            type="number"
            value="-48"
          />
        </div>
        <button
          data-testid="solve-btn"
          type="submit"
        >
          Solve
        </button>
      </form>
      <div>
        <h2>
          Solution
        </h2>
        <span>
          <span
            class="katex"
          >
            <math
              xmlns="http://www.w3.org/1998/Math/MathML"
            >
              <semantics>
                <mrow>
                  <msub>
                    <mi>
                      x
                    </mi>
                    <mrow>
                      <mn>
                        1
                      </mn>
                      <mo
                        separator="true"
                      >
                        ,
                      </mo>
                      <mn>
                        2
                      </mn>
                    </mrow>
                  </msub>
                  <mo>
                    =
                  </mo>
                  <mfrac>
                    <mrow>
                      <mo>
                        −
                      </mo>
                      <mi>
                        b
                      </mi>
                      <mo>
                        ±
                      </mo>
                      <msqrt>
                        <mrow>
                          <msup>
                            <mi>
                              b
                            </mi>
                            <mn>
                              2
                            </mn>
                          </msup>
                          <mo>
                            −
                          </mo>
                          <mn>
                            4
                          </mn>
                          <mo>
                            ∗
                          </mo>
                          <mi>
                            a
                          </mi>
                          <mo>
                            ∗
                          </mo>
                          <mi>
                            c
                          </mi>
                        </mrow>
                      </msqrt>
                    </mrow>
                    <mrow>
                      <mn>
                        2
                      </mn>
                      <mo>
                        ∗
                      </mo>
                      <mi>
                        a
                      </mi>
                    </mrow>
                  </mfrac>
                </mrow>
                <annotation
                  encoding="application/x-tex"
                >
                  x_{1,2}=\\frac{-b \\pm \\sqrt{b^2 - 4*a*c}}{2*a}
                </annotation>
              </semantics>
            </math>
          </span>
        </span>
        <span>
          <span
            class="katex"
          >
            <math
              xmlns="http://www.w3.org/1998/Math/MathML"
            >
              <semantics>
                <mrow>
                  <msub>
                    <mi>
                      x
                    </mi>
                    <mrow>
                      <mn>
                        1
                      </mn>
                      <mo
                        separator="true"
                      >
                        ,
                      </mo>
                      <mn>
                        2
                      </mn>
                    </mrow>
                  </msub>
                  <mo>
                    =
                  </mo>
                  <mfrac>
                    <mrow>
                      <mo>
                        −
                      </mo>
                      <mn>
                        8
                      </mn>
                      <mo>
                        ±
                      </mo>
                      <msqrt>
                        <mrow>
                          <msup>
                            <mn>
                              8
                            </mn>
                            <mn>
                              2
                            </mn>
                          </msup>
                          <mo>
                            −
                          </mo>
                          <mn>
                            4
                          </mn>
                          <mo>
                            ∗
                          </mo>
                          <mn>
                            1
                          </mn>
                          <mo>
                            ∗
                          </mo>
                          <mo>
                            −
                          </mo>
                          <mn>
                            48
                          </mn>
                        </mrow>
                      </msqrt>
                    </mrow>
                    <mrow>
                      <mn>
                        2
                      </mn>
                      <mo>
                        ∗
                      </mo>
                      <mn>
                        1
                      </mn>
                    </mrow>
                  </mfrac>
                </mrow>
                <annotation
                  encoding="application/x-tex"
                >
                  x_{1,2}=\\frac{-8 \\pm \\sqrt{8^2 - 4*1*-48}}{2*1}
                </annotation>
              </semantics>
            </math>
          </span>
        </span>
        <span>
          <span
            class="katex"
          >
            <math
              xmlns="http://www.w3.org/1998/Math/MathML"
            >
              <semantics>
                <mrow>
                  <msub>
                    <mi>
                      x
                    </mi>
                    <mrow>
                      <mn>
                        1
                      </mn>
                      <mo
                        separator="true"
                      >
                        ,
                      </mo>
                      <mn>
                        2
                      </mn>
                    </mrow>
                  </msub>
                  <mo>
                    =
                  </mo>
                  <mfrac>
                    <mrow>
                      <mo>
                        −
                      </mo>
                      <mn>
                        8
                      </mn>
                      <mo>
                        ±
                      </mo>
                      <msqrt>
                        <mn>
                          256
                        </mn>
                      </msqrt>
                    </mrow>
                    <mrow>
                      <mn>
                        2
                      </mn>
                      <mo>
                        ∗
                      </mo>
                      <mn>
                        1
                      </mn>
                    </mrow>
                  </mfrac>
                </mrow>
                <annotation
                  encoding="application/x-tex"
                >
                  x_{1,2}=\\frac{-8 \\pm \\sqrt{256}}{2*1}
                </annotation>
              </semantics>
            </math>
          </span>
        </span>
        <span>
          x1 =
          <span
            data-testid="root-x1"
          >
            4
          </span>
        </span>
        <span>
          x2 =
          <span
            data-testid="root-x2"
          >
            -12
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
